pandas/compat/numpy/function.py

```diff
@@ -155,13 +158,13 @@ def validate_clip_with_axis(axis, args, kwargs):
     return axis
 
 
-COMPRESS_DEFAULTS = OrderedDict()
+COMPRESS_DEFAULTS = OrderedDict()  # type: OrderedDict[str, None]
 COMPRESS_DEFAULTS['axis'] = None
 COMPRESS_DEFAULTS['out'] = None
 validate_compress = CompatValidator(COMPRESS_DEFAULTS, fname='compress',
                                     method='both', max_fname_arg_count=1)
 
-CUM_FUNC_DEFAULTS = OrderedDict()
+CUM_FUNC_DEFAULTS = OrderedDict()  # type: OrderedDict[str, None]
```

Comment:
Just a random callout especially for @gfyoung since we had a similar conversation yesterday, but I think there's a nuance to how we use the actual type of OrderedDict versus the generic type in the typing module (added recently in 3.7.2).

In this annotation it doesn't cause problems, but if this was in the signature of the function I think we would be getting a syntax error:

https://stackoverflow.com/questions/41207128/how-do-i-specify-ordereddict-k-v-types-for-mypy-type-annotation

Not sure we need to do anything yet just providing as FYI

Reply:
Indeed, that's true.  We'll have to keep an eye out for that as we slowly add more of the annotations to the codebase.

Reply:
@WillAyd unrelated to typing but why do we take 3 lines to create and assign values to `CUM_FUNC_DEFAULTS`(and other similar variables) when it can be easily done in a single line. Upper case name indicates it's a constant, so shouldn't we do something like this:

    CUM_FUNC_DEFAULTS = OrderedDict(dtype=None, out=None)

Reply:
@vaibhavhrt : The 3 lines are to ensure that keys are inserted exactly in the order that we want them to.  If you use the one-liner, earlier Python 3.x versions would insert them based on Python's natural ordering of objects.

We probably could eventually replace with a one-liner, since later Python 3.x versions fix this issue, but I wouldn't do it immediately.

Reply:
@gfyoung thanks for explaining.